lib/render.js

```diff
@@ -25,28 +27,39 @@ function composeData(data) {
 }
 
 /**
- * Take in the data returned from Amphora and turn it into HTML output.
+ * Take in the data returned from Amphora, call `render` hooks,
+ * and return HTML output.
  *
  * @param  {Object} data
- * @return {Promise}
+ * @return {Promise<String>} html
  */
 function render(data) {
-  const template = clayUtils.getComponentName(data._layoutRef || data._self),
-    rootPartial = hbs.partials[template],
-    composedData = composeData(data);
-
-  if (!rootPartial) {
-    throw new Error(`Missing template for ${data.template}`);
-  }
-
-  return Promise.resolve(rootPartial(composedData))
-    .then(mediaService.append(data._media, data.locals.edit, data.locals.site))
-    .then(result => {
-      return {
-        output: result,
-        type: 'html'
-      };
-    });
+
+  const hooks = _.chain(setup.plugins)
+                  .map('render')
+                  .compact()
+                  .value();
+
+  return bluebird.reduce(hooks, (data, hook) => hook(data), data)
+  .then(data => {
+    // console.log(JSON.stringify(data, null, 2));
+    const template = clayUtils.getComponentName(data._layoutRef || data._self),
+      rootPartial = hbs.partials[template],
+      composedData = composeData(data);
+
+    if (!rootPartial) {
+      throw new Error(`Missing template for ${data.template}`);
+    }
+
+    return rootPartial(composedData);
+  })
```

Comment:
Based on the complexity of hooks, I'd maybe see about separating this out into its own function

Reply:
Fixed ✔️